core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -628,6 +621,13 @@ private[spark] class ExecutorAllocationManager(
             allocationManager.onExecutorIdle(executorId)
           }
         }
+
+        // If the task failed, we expect it to be resubmitted later.
+        if (taskEnd.reason != Success) {
+          stageIdToTaskIndices.get(stageId).foreach { taskIndices =>
+            taskIndices.remove(taskIndex)
```

Comment:
any reason why you didn't use all of Andrew's suggestion earlier?  specifically:

```scala
    val numTasksTotal = stageIdToNumTasks.getOrElse(stageId, -1)
    if (taskIndices.size == numTasksTotal) {
      allocationManager.onSchedulerBacklogged()
    }
    taskIndices.remove(taskIndex)
  }

```